### PR TITLE
Live commit on multi-disks

### PR DESCRIPTION
**Pull request by aliang123**
split blockdev_commit_base from blockdev_commit
support block commit on multi-disks

Signed-off-by: Aihua Liang <aliang@redhat.com>

id:1817011

### CODE REVIEW

**Comment by aliang123** [comment]
(1/2) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.blockdev_commit.q35: PASS (76.26 s)
 (2/2) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.blockdev_commit.multi_data_disks.q35: PASS (98.45 s)

**Comment by aliang123** [comment]
@ngu-niny, @xutian , please help to review, thanks

**Comment by aliang123** [comment]
(1/2) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.blockdev_commit.q35: PASS (79.05 s)
 (2/2) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.blockdev_commit.multi_data_disks.q35: PASS (98.42 s)

**Comment by ngu-niny** [comment]
LGTM.

Test PASS:

'1-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.ppc64le.io-github-autotest-qemu.blockdev_commit' Finshed 2020-04-01 21:41:12 2020-04-01 21:43:36 PASS 144.185782
'2-Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.ppc64le.io-github-autotest-qemu.blockdev_commit.multi_data_disks' Finshed 2020-04-01 21:43:36 2020-04-01 21:46:36 PASS 179.93927

Acked-by: Nini Gu <ngu@redhat.com>

File: provider/blockdev_commit_base.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
+import logging
```
Comment (zhencliu) [inline provider/blockdev_commit_base.py:1]: Put BlockDevCommitTest into provider, then I guess more test cases besides blockdev_commit.py will employ BlockDevCommitTest to run tests in future, right?

Comment (aliang123) [reply provider/blockdev_commit_base.py:1]: Yes, you're right.

File: provider/blockdev_commit_base.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    def post_test(self):
+        try:
+            self.main_vm.destroy()
+            for image in self.snapshot_images:
+                image.remove()
+        except Exception as error:
+            logging.error(str(error))
+
+    def run_test(self):
+        self.pre_test()
```
Comment (zhencliu) [inline provider/blockdev_commit_base.py:148]: Please add 'kill_vm = yes' in cfg file in case some error might occur in pre_test

Comment (aliang123) [reply provider/blockdev_commit_base.py:148]: OK, will update it.

**Comment by aliang123** [comment]
@zhencliu, have updated, please help to review again.

**Comment by zhencliu** [comment]
LGTM

**Comment by aliang123** [comment]
Thanks, @lolyu, @ngu-niny and @zhencliu

**Pull request merged**